### PR TITLE
Add GID-15 Dataset

### LINKED ISSUE #122: Add GID-15 Dataset
We should add the [GID-15 dataset](https://captain-whu.github.io/GID15/) from ["Land-Cover Classification with High-Resolution Remote Sensing Images Using Transferable Deep Models", Tong et al. (2018)"](https://arxiv.org/abs/1807.05713). It has a good representation of imagery over China taken by the Gaofen-2 satellite for semantic segmentation.

### PR DESCRIPTION
**Pull request by isaaccorley**
- Adds `torchgeo.datasets.GID15`
- Updated docs
- Added `tests/datasets/test_gid15.py`
- Added sample data for tests under `tests/data/gid15/`

Closes #122

### CODE REVIEW

File: torchgeo/datasets/gid15.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        7. irrigated_land
+        8. dry_cropland
+        9. garden_plot
+        10. arbor_woodland
+        11. shrub_land
+        12. natural_grassland
+        13. artificial_grassland
+        14. river
+        15. lake
+        16. pond
```
Comment (adamjstewart) [inline torchgeo/datasets/gid15.py:52]: I would remove the extra indentation from these lines

File: torchgeo/datasets/gid15.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        image_root = os.path.join(root, "GID", "img_dir")
+        images = glob.glob(os.path.join(image_root, split, "*.tif"))
+        images = sorted(images)
+        if split != "test":
+            masks = [
+                image.replace("img_dir", "ann_dir").replace(".tif", "_15label.png")
+                for image in images
+            ]
+        else:
+            masks = [""] * len(images)
```
Comment (adamjstewart) [inline torchgeo/datasets/gid15.py:173]: Why is this needed?

File: torchgeo/datasets/gid15.py (hunk 1 above)
Comment (isaaccorley) [reply torchgeo/datasets/gid15.py]: Fixed

File: torchgeo/datasets/gid15.py (hunk 2 above)
Comment (isaaccorley) [reply torchgeo/datasets/gid15.py]: It's not. I simply removed the masks from the file lists for the test set now.

**Review by adamjstewart — APPROVED** [review]
